pennylane/gradients/finite_difference.py

```diff
@@ -359,3 +359,255 @@ def processing_fn(results):
+        [[-0.38751721 -0.18884787 -0.38355704]
+         [ 0.69916862  0.34072424  0.69202359]]
+    """
+    if argnum is None and not tape.trainable_params:
+        warnings.warn(
+            "Attempted to compute the gradient of a tape with no trainable parameters. "
+            "If this is unintended, please mark trainable parameters in accordance with the "
+            "chosen auto differentiation framework, or via the 'tape.trainable_params' property."
+        )
+        return [], lambda _: qml.math.zeros([tape.output_dim, 0])
+
+    if validate_params:
+        if "grad_method" not in tape._par_info[0]:
+            gradient_analysis(tape, grad_fn=finite_diff_new)
+        diff_methods = grad_method_validation("numeric", tape)
+    else:
+        diff_methods = ["F" for i in tape.trainable_params]
+
+    if all(g == "0" for g in diff_methods):
+        output_dims = []
+        for m in tape.measurements:
+            if m.return_type is qml.measurements.Probability:
+                output_dims.append(2 ** len(m.wires))
+            else:
+                output_dims.append(1)
+        list_zeros = []
+
+        for i, _ in enumerate(tape.measurements):
+            dim = output_dims[i]
+            sub_list_zeros = []
+            for _ in range(0, len(tape.trainable_params)):
+                sub_list_zeros.append(qml.math.zeros(dim))
+            sub_list_zeros = tuple(sub_list_zeros)
+            list_zeros.append(sub_list_zeros)
+
+        if len(tape.measurements) == 1:
+            return [], lambda _: list_zeros[0]
+
+        return [], lambda _: tuple(list_zeros)
+
+    gradient_tapes = []
+    shapes = []
+    c0 = None
+
+    coeffs, shifts = finite_diff_coeffs(n=n, approx_order=approx_order, strategy=strategy)
+
+    if 0 in shifts:
+        # Finite difference formula includes a term with zero shift.
+
+        if f0 is None:
+            # Ensure that the unshifted tape is appended
+            # to the gradient tapes, if not already.
+            gradient_tapes.append(tape)
+
+        # Store the unshifted coefficient. We know that
+        # it will always be the first coefficient due to processing.
+        c0 = coeffs[0]
+        shifts = shifts[1:]
+        coeffs = coeffs[1:]
+
+    method_map = choose_grad_methods(diff_methods, argnum)
+
+    for i, _ in enumerate(tape.trainable_params):
+        if i not in method_map or method_map[i] == "0":
+            # parameter has zero gradient
+            shapes.append(0)
+            continue
+
+        g_tapes = generate_shifted_tapes(tape, i, shifts * h)
+        gradient_tapes.extend(g_tapes)
+        shapes.append(len(g_tapes))
+
+    def processing_fn(results):
+        grads = []
+        start = 1 if c0 is not None and f0 is None else 0
+        r0 = f0 or results[0]
+
+        for s in shapes:
+
+            if s == 0:
+                # parameter has zero gradient
+                g = qml.math.zeros_like(results[0])

```

Comment:
Is there a bug here? If the tape has multiple measurements, then `results[0]` will be a tuple (which is an invalid argument to `zeros_like`.

Reply:
Thank you for point this out I am will check that!

Reply:
Great catch, I updated the function and added a test 💯